cmd/collector/app/handler/otlp_receiver.go

```diff
@@ -106,7 +104,7 @@ func applyGRPCSettings(cfg *configgrpc.ServerConfig, opts *flags.GRPCOptions) {
 		cfg.NetAddr.Endpoint = opts.HostPort
 	}
 	if opts.TLS.Enabled {
-		cfg.TLSSetting = applyTLSSettings(&opts.TLS)
+		cfg.TLSSetting = opts.TLS.ToOtelServerConfig()
```

Comment:
the approach in the other PRs is to replace the config field type with `configtls`, can we use that here too?

Reply:
If we use configtls.ServerConfig here https://github.com/jaegertracing/jaeger/blob/641ddf2860288975518c37c3efdac2761d39b0a6/cmd/collector/app/flags/flags.go#L145 but the config has no enabled field or insecure field like clientConfig . Then how should i deal with opts.TLS.Enabled

Reply:
`ToOtelServerConfig `just returns nil if tls is not enabled , This will work .

Reply:
> Then how should i deal with opts.TLS.Enabled

we should deal with it the same way as in the other PRs. The interface in v1 is CLI flags, it doesn't matter if we translate them into internal TLS config of the one from OTEL.